src/lib/y2partitioner/widgets/format_and_mount.rb

```diff
@@ -177,143 +230,183 @@ def mount?
+
+      def opt
+        %i(hstretch notify)
       end
 
       def init
-        self.value = @filesystem
+        fs_type = @options.filesystem_type
+        self.value = fs_type ? fs_type.to_sym : nil
       end
 
       def label
         _("Filesystem")
       end
 
       def items
-        Y2Storage::Filesystems::Type.all.map do |fs|
-          [fs.to_s, fs.to_human_string]
+        return [] unless @options.filesystem_type
+
+        Y2Storage::Filesystems::Type.all.select { |fs| supported?(fs) }.map do |fs|
+          [fs.to_sym, fs.to_human_string]
         end
       end
 
+      def supported?(fs)
+        %i(swap btrfs ext2 ext3 ext4 vfat xfs reiserfs).include?(fs.to_sym)
+      end
+
       def store
-        @filesystem = value
+        @options.filesystem_type = value ? Y2Storage::Filesystems::Type.find(value) : value
       end
+    end
 
-      def selected_filesystem
-        Y2Storage::Filesystems::Type.all.detect do |fs|
-          fs.to_s == value
-        end
+    # Push Button that launches a dialog to set speficic options for the
+    # selected filesystem
+    class FormatOptionsButton < CWM::PushButton
+      def initialize(options)
+        @options = options
+      end
+
+      def opt
+        %i(hstretch notify)
+      end
+
+      def label
+        _("Options...")
+      end
+
+      def handle
+        # Dialogs::FormatOptions.new(@options).run
+
+        nil
       end
     end
 
     # MountPoint selector
     class MountPoint < CWM::ComboBox
-      def initialize(mount_point)
-        @mount_point = mount_point
+      def initialize(options)
+        @options = options
       end
 
       def init
-        self.value = @mount_point
+        self.value = @options.mount_point
       end
 
       def label
         _("Mount Point")
       end
 
       def opt
-        [:editable, :hstretch, :notify]
+        %i(editable hstretch notify)
       end
 
       def store
-        @mount_point = value
+        @options.mount_point = value
       end
 
       def items
-        %w(/root /home /opt /var).map { |mp| [mp, mp] }
+        %w(/root /home /srv /tmp /opt /var).map { |mp| [mp, mp] }
       end
     end
 
     # Encryption selector
     class EncryptBlkDevice < CWM::CheckBox
-      def initialize(encrypt)
-        @encrypt = encrypt
+      def initialize(options)
+        @options = options
       end
 
       def label
         _("Encrypt Device")
       end
 
       def init
-        self.value = @encrypt
+        if @options.filesystem_type && !@options.filesystem_type.encryptable?
+          self.value = false
+          disable
+        else
+          self.value = @options.encrypt
+        end
       end
 
       def store
-        @encrypt = value
+        @options.encrypt = value
       end
     end
 
-    # FIME(WIP)
-    class FstabOptionsButton < CWM::PushButton
+    # Inode Size format option
+    class InodeSize < CWM::ComboBox
+      SIZES = ["auto", "512", "1024", "2048", "4096"].freeze
```

Comment:
I am not sure if UI should drive available options...and BTW auto is not translated.